src/style_layer.js

```diff
@@ -0,0 +1,88 @@
+'use strict';
+
+var util = require('./util');
+var format_url = require('./format_url');
+
+var StyleLayer = L.TileLayer.extend({
+    includes: [require('./load_tilejson')],
+
+    options: {
+        sanitizer: require('sanitize-caja')
+    },
+
+    scalePrefix: '@2x',
+
+    initialize: function(_, options) {
+        L.TileLayer.prototype.initialize.call(this, undefined, options);
+
+        this.options.tiles = this._formatTileURL(_);
+
+        this._tilejson = {};
+
+        if (options && options.format) {
+            util.strict_oneof(options.format, this.formats);
+        }
+
+        this._loadTileJSON(_);
+    },
+
+    _setTileJSON: function(json) {
+        util.strict(json, 'object');
+
+        L.extend(this.options, {
+            attribution: this.options.sanitizer(json.attribution),
+            minZoom: json.minzoom || 0,
+            maxZoom: json.maxzoom || 22,
+            bounds: json.bounds && util.lbounds(json.bounds),
+            tms: false,
+            tileSize: 512,
+            zoomOffset: -1
+        });
+
+        this._tilejson = json;
+        this.redraw();
```

Comment:
This will be fixed by https://github.com/mapbox/mapbox.js/pull/1108/files#r50455207. `this.redraw()` removes and readds all tiles, but we only need to refresh attribution.